src/sentry/rules/conditions/tagged_event.py

```diff
@@ -32,6 +35,17 @@ class MatchType(object):
 )
 
 
+def get_project_release_cache_key(project_id):
+    return u"project:{}:latest_release".format(project_id)
+
+
+@release_created.connect(weak=False)
+def clear_project_release_cache(release, **kwargs):
+    release_project_ids = release.projects.values_list("id", flat=True)
+    for proj_id in release_project_ids:
+        cache.delete(get_project_release_cache_key(proj_id))
```

Comment:
We can use `delete_many` here instead so that we don't need N calls to the cache.